asserts/database.go

```diff
@@ -27,22 +27,34 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
+	"time"
 
 	"golang.org/x/crypto/openpgp/packet"
 
 	"github.com/ubuntu-core/snappy/helpers"
 )
 
+// PublicKey is a public key as used by the assertion database.
+type PublicKey interface {
+	PublicKeyPrim
```

Comment:
@niemeyer this probably me jumping ahead by half, and I'm not happy with the naming but as mention in a TODO in account_key.go itself, AccountKey would wrap something with the functionality of PublicKeyPrim produced by parsing public key functionaly in crypto.go and the whole of AccountKey would implement PublicKey which has the IsValidAt as well (which would be implemented using since and until)